src/pydicom/pixels/decoders/native.py

```diff
@@ -64,10 +65,14 @@ def _decode_frame(src: bytes, runner: DecodeRunner) -> bytearray:
         # Update the runner options to ensure the reshaping is correct
         # Only do this if we successfully decoded the frame
         runner.set_option("planar_configuration", 1)
+    else:
+        frame = _deflated_decode_frame(src)
 
-        return frame
+    # Unpack bit-packed data
+    if runner.bits_allocated == 1:
+        frame = bytearray(unpack_bits(frame))
```

Comment:
`unpack_bits(frame, as_array=False)`

Reply:
resolved in [c1e9a0f](https://github.com/pydicom/pydicom/pull/2216/commits/c1e9a0f2c1b8225b608765a11be870fdd5d7fc99)